// ez-components/dist/esm/my-accordion.entry.js
import { r as registerInstance, c as createEvent, h } from './index-87bf0dfa.js';

const myAccordionScss = "* {\n    font-family: 'Lato', sans-serif;\n}\n\n.accordion {\n    cursor: pointer;\n    padding: 18px;\n    text-align: left;\n    border-radius: 20px;\n    font-size: 1.2rem;\n    font-weight: bold;\n    outline: 0;\n    span {\n        float: right;\n    }\n}\n \n .open {\n    display: block;\n    height: auto;\n    border-radius: 20px;\n    border: 0.5px solid rgb(199, 197, 197);\n    padding: 18px;\n }\n\n .close {\n    display: none;\n }\n";

const MyComponent = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.toggle = false;
        this.onToggle = createEvent(this, "onToggle", 7);
    }
    toggleComponent() {
        this.toggle = !this.toggle;
        this.onToggle.emit({ visible: this.toggle });
    }
    render() {
        return (h("div", null, h("button", { class: "accordion", style: {
                width: this.width,
                backgroundColor: this.color,
            }, onClick: () => this.toggleComponent() }, this.label, this.toggle ? h("span", null, "\u25B2") : h("span", null, "\u25BC")), h("div", { class: `content-box ${this.toggle ? 'open' : 'close'}`, style: { width: this.width } }, h("p", null, this.description))));
    }
};
MyComponent.style = myAccordionScss;

export { MyComponent as my_accordion };
