// ez-components/dist/cjs/my-component.cjs.entry.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-36266b65.js');

const myComponentScss = "* {\n  font-family: 'Lato', sans-serif;\n}\n\n.container {\n  display: -ms-flexbox;\n  display: flex;\n  -ms-flex-direction: column;\n  flex-direction: column;\n  -ms-flex-pack: center;\n  justify-content: center;\n  -ms-flex-align: center;\n  align-items: center\n}\n\n.accordion {\n  cursor: pointer;\n  padding: 18px;\n  text-align: left;\n  border-radius: 20px;\n  font-size: 1.2rem;\n  font-weight: bold;\n  outline: 0;\n  span {\n      float: right;\n  }\n}\n\n.open {\n  display: block;\n  height: auto;\n  border-radius: 20px;\n  border: 0.5px solid rgb(199, 197, 197);\n  width: 200px;\n}\n\np {\n  padding: 18px;\n}\n\n.close {\n  display: none;\n}";

const MyComponent = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.toggle = false;
        this.onToggle = index.createEvent(this, "onToggle", 7);
    }
    toggleComponent() {
        this.toggle = !this.toggle;
        this.onToggle.emit({ visible: this.toggle });
    }
    render() {
        return (index.h("div", null, index.h("button", { class: "accordion", style: {
                width: this.width,
                backgroundColor: this.color,
            }, onClick: () => this.toggleComponent() }, this.label, this.toggle ? index.h("span", null, "\u25B2") : index.h("span", null, "\u25BC")), index.h("div", { class: `content-box ${this.toggle ? 'open' : 'close'}`, style: { width: this.width } }, index.h("p", null, this.description))));
    }
};
MyComponent.style = myComponentScss;

exports.my_component = MyComponent;
